### DISCUSSION TITLE
problem with overflow + break-down text

### DISCUSSION THREAD

**Discussion by mxcdh** [tailwindlabs/tailwindcss#9106] [Help]
Hello

I using tailwind-overflow, for horizontal scroll in right side.
Bellow example of my code:
```
  <div className="bg-red-300">
        <div className="bg-red-400">
          <div className="container mx-auto pl-3 pr-3 pt-6 pb-6 flex space-x-2  overflow-x-auto">
            <div className="flex space-x-2">
              <div>Explore:</div>
              <div className="font-bold underline">Shoes</div>
              <div className="font-bold underline">Clothing</div>
              <div className="font-bold underline">Accessories</div>
              <div className="font-bold underline">Premium</div>
              <div className="font-bold underline">Sport</div>
              <div className="font-bold underline">Shop All</div>
            </div>
          </div>
        </div>
      </div>
```

#1 Why word Shop All was break, when screen is mobile? How fix is it?
- screenshot: https://i.imgur.com/QpOFXpn.gif
- demo: https://superb-pothos-b7b687.netlify.app/tailwind-overflow
#2 How I can hide scroll in tailwind?

**Comment by wongjn** [comment]
This happens because the elements in flex will shrink as much in width as possible (up to `min-content`) due to [`flex-shrink` being `1` by default](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-shrink#formal_definition) before "giving in" to the `overflow-x: auto`. If we look at the [definition of `min-content`](https://developer.mozilla.org/en-US/docs/Web/CSS/min-content):

> The `min-content` sizing keyword represents the intrinsic minimum width of the content. For text content this means that the content will take all soft-wrapping opportunities, becoming as small as the longest word.

So there is a soft-wrap opportunity in "Shop All" because it is made of two words.

If you wish to avoid this wrapping, one can set the element to not shrink, by setting `flex-shrink: 0` using `shrink-0` utility class:

```diff
   <div className="font-bold underline">Sport</div>
-  <div className="font-bold underline">Shop All</div>
+  <div className="font-bold underline shrink-0">Shop All</div>
 </div>
```

[See this working in this Tailwind Play](https://play.tailwindcss.com/XkKJ6cbhQb).

**Reply by mxcdh** [reply]
@wongjn  thank you, I have one question, is it possible to remove/hide the scroll?
https://drive.google.com/file/d/15HvizjhwpKmxlhjNGogSynq_VOnhpdf0/view

p.s
when I use  <div className=" flex space-x-2  overflow-x-auto overflow-x-hidden"> I hidden scroll, but I can scroll to right. I want a hidden scroll, but I want to scroll when my mouse is on the text and scroll right on the touchpad.

**Reply by wongjn** [reply]
You can set [`scrollbar-width` to `none`](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-width#values) with `[scrollbar-width:none]` for Firefox but beware of [accessibility concerns](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-width#accessibility_concerns): 

> setting `scrollbar-width` to `thin` or `none` can make content hard or impossible to scroll if the author does not provide an alternative scrolling mechanism. While swiping gestures or mouse wheels can enable scrolling on such content, some devices have no scroll alternative.

For other browsers, you could set the [`::-webkit-scrollbar` pseudo element](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar) to `display: none` with `[&::-webkit-scrollbar]:hidden`.
